src/app.js

```diff
@@ -0,0 +1,218 @@
+/*global $:false, Game, Frame */
+
+$(document).ready(function() {
+  var game, gameNumber, rollValue;
+
+  gameNumber = 0;
+  startNewGame();
+
+  $( 'body' ).on( 'click', '#startNewGame', function() {
+   $( '#userInterface > p' ).html( 'Click the number of pins to knock down' );
+   game = new Game();
+   gameNumber++;
+   drawButtons();
+   drawScoreCard();
+  });
+
+  $( 'body' ).on( 'click', '.btn', function() {
+    rollValue = $(this).attr('data-value');
+    game.roll( parseInt(rollValue) );
+    updateScore();
+
+    if( game.isGameOver() ) {
+      startNewGame();
+    } else {
+      drawButtons();
+    }
+  })
+
+  function startNewGame() {
+    var newGame = '<button id="startNewGame" class="game-btn" type="button" ' +
+                  'data-value="Play a new Game">Play a new Game</button>';
+   $( '#buttonList' ).empty();
+   $( '#userInterface > p' ).html(newGame);
+  }
+
+  function drawButtons() {
+    var i,
+        btn,
+        numButtons,
+        currentFrame = game.currentFrame(),
+        pins = currentFrame.pins;
+
+    if( isLastFrameReset() ) {
+      numButtons = 10;
+    } else {
+      numButtons = pins < 10 ? 10 - pins : 20 - pins;
+    }
+
+    $( '#buttonList' ).empty();
+    for( i = 0; i <= numButtons; i++ ) {
+      btn  =  "<button id='btn" + i + "' type='button' class='btn' " +
+              "data-value='" + i + "'>" + i + "</button>";
+      $( '#buttonList' ).append(btn);
+    }
+  }
+
+  function isLastFrameReset() {
+    var currentFrame = game.currentFrame();
+
+    return currentFrame.isLastFrame() && ( currentFrame.isStrike() ||
+           currentFrame.pins === 20 || currentFrame.isSpare() );
+  }
+
+  function updateScore() {
+    game.frames.forEach( function( frame, index ) {
```

Comment:
This function has too many statements. (15)<br>This function's cyclomatic complexity is too high. (8)